src/cljs/org/broadinstitute/firecloud_ui/page/workspaces.cljs

```diff
@@ -153,36 +153,42 @@
    :width "100%" :fontSize "88%"})
 
 (defn- render-overlay [state refs]
-  [:div {:style (modal-background state)}
-   [:div {:style modal-content}
-    [:div {:style {:backgroundColor "#fff"
-                   :borderBottom "1px solid #e3e3e3"
-                   :padding "20px 48px 18px"
-                   :fontSize "137%" :fontWeight 400 :lineHeight 1}}
-     "Create New Workspace"]
-    [:div {:style {:padding "22px 48px 40px" :boxSizing "inherit"}}
-     [:div {:style form-label} "Name Your Workspace"]
-     [:input {:type "text" :style new-ws-input-style :ref "wsName"}]
-     [:div {:style form-label} "Workspace Description"]
-     [:textarea {:style new-ws-input-style :rows 10 :cols 30 :ref "wsDesc"}]
-     [:div {:style form-label} "Research Purpose"]
-     [:select {:style select} [:option {} "Option 1"] [:option {} "Option 2"] [:option {} "Option 3"]]
-     [:div {:style form-label} "Billing Contact"]
-     [:select {:style select} [:option {} "Option 1"] [:option {} "Option 2"] [:option {} "Option 3"]]
-     [:div {:style form-label} "Share With (optional)"]
-     [:input {:type "text" :style new-ws-input-style :ref "shareWith"}]
-     [:em {:style {:fontSize "69%"}} "Separate multiple emails with commas"]
-     (let [clear-overlay (fn []
-                           (set! (.-value (.getDOMNode (@refs "wsName"))) "")
-                           (set! (.-value (.getDOMNode (@refs "wsDesc"))) "")
-                           (set! (.-value (.getDOMNode (@refs "shareWith"))) "")
-                           (swap! state assoc :overlay-shown? false))]
+  (let [clear-overlay (fn []
+                        (set! (.-value (.getDOMNode (@refs "wsName"))) "")
+                        (set! (.-value (.getDOMNode (@refs "wsDesc"))) "")
+                        (set! (.-value (.getDOMNode (@refs "shareWith"))) "")
+                        (swap! state assoc :overlay-shown? false))]
+    [:div {:style (modal-background state)
+           :onKeyDown (fn [e] (when (= 27 (.-keyCode e)) (clear-overlay)))}
```

Comment:
Consider something like this for clarity (in every case where you do this comparison):
```
(when (= (:esc common/key-codes) (.-keyCode e))
  ...
```